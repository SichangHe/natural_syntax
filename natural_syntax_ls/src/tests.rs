use super::*;

use insta::assert_debug_snapshot;
use tracing::Level;
use tracing_subscriber::EnvFilter;

#[test]
fn token_types() {
    init_tracing();
    let types = semantic_token_types();
    assert_debug_snapshot!(types);
}

#[test]
fn token_modifiers() {
    init_tracing();
    let modifiers = semantic_token_modifiers();
    assert_debug_snapshot!(modifiers);
}

fn init_tracing() {
    _ = tracing_subscriber::fmt()
        .with_env_filter(
            EnvFilter::builder()
                .with_default_directive(Level::INFO.into())
                .from_env_lossy(),
        )
        .try_init();
}
