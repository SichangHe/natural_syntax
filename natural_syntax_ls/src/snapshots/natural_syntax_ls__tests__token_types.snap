---
source: natural_syntax_ls/src/tests.rs
expression: types
---
[
    SemanticTokenType(
        "namespace",
    ),
    SemanticTokenType(
        "type",
    ),
    SemanticTokenType(
        "class",
    ),
    SemanticTokenType(
        "enum",
    ),
    SemanticTokenType(
        "interface",
    ),
    SemanticTokenType(
        "struct",
    ),
    SemanticTokenType(
        "typeParameter",
    ),
    SemanticTokenType(
        "parameter",
    ),
    SemanticTokenType(
        "variable",
    ),
    SemanticTokenType(
        "property",
    ),
    SemanticTokenType(
        "enumMember",
    ),
    SemanticTokenType(
        "event",
    ),
    SemanticTokenType(
        "function",
    ),
    SemanticTokenType(
        "method",
    ),
    SemanticTokenType(
        "macro",
    ),
    SemanticTokenType(
        "keyword",
    ),
    SemanticTokenType(
        "modifier",
    ),
    SemanticTokenType(
        "comment",
    ),
    SemanticTokenType(
        "string",
    ),
    SemanticTokenType(
        "number",
    ),
    SemanticTokenType(
        "regexp",
    ),
    SemanticTokenType(
        "operator",
    ),
    SemanticTokenType(
        "decorator",
    ),
]
